homeassistant/components/airzone/select.py

```diff
@@ -50,6 +67,24 @@ class AirzoneSelectDescription(SelectEntityDescription):
 }
 
 
+def main_zone_options(entity: AirzoneEntity, options: dict[str, int]) -> dict[str, int]:
+    """Filter available modes."""
+    modes = entity.get_airzone_value(AZD_MODES)
+    return {k: v for k, v in options.items() if v in modes}
+
+
+MAIN_ZONE_SELECT_TYPES: Final[tuple[AirzoneSelectDescription, ...]] = (
+    AirzoneSelectDescription(
+        api_param=API_MODE,
+        key=AZD_MODE,
+        options=list(MODE_DICT),
```

Comment:
If we're going to use a function for the options, we should omit this one

Reply:
Done in 38e6f6b57b18eae688f19f737c13f377547b4539